### ISSUE TITLE
[BUG] Handle language server is closed error in user readable format

### ISSUE THREAD

**Issue by shawn-hurley** [konveyor/analyzer-lsp#500]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Konveyor version

All

### Priority

Minor

### Current Behavior

When we get an error from a language server, we usually get it from an RPC connection either stopping or being unreachable. The current errors that come out of the package are like: ```ERRO[0015] unable to ask for tackle rule entry           error="write |1: broken pipe" provider=java```

### Expected Behavior

We should expose errors that make sense to the end user and log the errors coming from RPC at a lower verbosity for debugging. 

The error that is exposed should be "connection to the language server is closed, language server is not running"

### How Reproducible

Always (Default)

### Steps To Reproduce

1. podman build -f demo.Dockerfile
2. Podman run -it -m 1G --entrypoint /bin/bash <image-sha>
3. konveyor-analyzer

The pod is killed, and you will eventually see: ERRO[0015] unable to ask for tackle rule entry           error="write |1: broken pipe" provider=java

### Environment

```markdown
- **Analyzer Image SHA** 
- **JDTLS  base image SHA**
- **Provider Settings JSON**
```

### Anything else?

To fix this, we will need to capture this error from the RPC client and override it. 

First find all the times we call `rpc.Call` function. Create a new package in `jsonrpc2` called `rpcerr`. 

From here create a function that will look like: 

```go 
func IsRPCClosed(err error) bool 
```

This function will compare the error string, for the error that we don't want to see. If it finds it return true, if not return false.

Where we deal with the rpc.Call error, add a block that will see if it is a closed error, if it is, change the error in the log to: 
``fmt.Errrof("connection to the language server is closed, lanaguage server is not running")```

**Comment by jmle** [comment]
Done by @abrugaro
